compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/PropertyReferenceLowering.kt

```diff
@@ -95,6 +99,33 @@ internal class PropertyReferenceLowering(val context: JvmBackendContext) : Class
             calculateOwner(expression.propertyContainer, this@PropertyReferenceLowering.context)
         }
 
+    private fun IrBuilderWithScope.computeSignatureString(expression: IrCallableReference): IrExpression {
+        return expression.getter?.let {
+            localPropertyIndices[it]?.let { irString("<v#$it>") }
+                ?: if (it.owner.origin == IrDeclarationOrigin.DEFAULT_PROPERTY_ACCESSOR) {
+                    // Default property accessor. Compute the signature now, so that we will not get into trouble
+                    // if the getter is transformed to a static method by inline classes lowering.
+                    irString(expression.signature)
```

Comment:
I suppose it'd be useful to inline the `signature` extension property here, since it also contains the logic about local variable references (`<v#...>`), which is not needed anymore here since by this point it's proven to be null

Reply:
Yes, that is better. Inlined.

Reply:
And now it is actually inlined and simplified properly. Thanks.